### PR TITLE
Enable `check_untyped_defs` in `tests/integration/transfer`

### PR DESCRIPTION
**Pull request by karlb**
The next step will be to enable it for all integration tests except those in `raiden/tests/integration/network`. 

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/tests/utils/events.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
 from raiden.raiden_service import RaidenService
 from raiden.storage.sqlite import RANGE_ALL_STATE_CHANGES
 from raiden.transfer.architecture import Event, StateChange
 from raiden.transfer.mediated_transfer.events import EventUnlockClaimFailed, EventUnlockFailed
-from raiden.utils.typing import Any, Iterable, List, Optional, Tuple, Type, TypeVar
 
 NOVALUE = object()
 T = TypeVar("T")
+T_Event = TypeVar("T_Event", bound="Event")
```
Comment (palango) [inline raiden/tests/utils/events.py]: Not sure how this should be named, I think we use`T_*` usually for the underlying type. Also in line 15 a non `T_` prefixed name is used.

File: raiden/tests/utils/transfer.py [hunk 2]
```diff
@@ -1075,5 +1075,7 @@ def block_timeout_for_transfer_by_secrethash(
     )
 
 
-def calculate_fee_for_amount(amount: int) -> int:
-    return round((amount * INTERNAL_ROUTING_DEFAULT_FEE_PERC) * (1 + DEFAULT_MEDIATION_FEE_MARGIN))
+def calculate_fee_for_amount(amount: int) -> FeeAmount:
```
[the remark this answers is not in the document]
Comment (karlb) [reply raiden/tests/utils/transfer.py]: I didn't want to introduce new typing errors in this PR, so I didn't touch more than necessary and focused on resolving the existing errors.

File: raiden/tests/utils/events.py (hunk 1 above)
Comment (karlb) [reply raiden/tests/utils/events.py]: For the `StateChange` we already have one defined below, called `SC`. I should remove that line. I could rename `T_Event` to `TE`, which would be in line with `TM` below. Not the best names, but at least consistent.

Comment (palango) [reply raiden/tests/utils/events.py]: Sounds good to me.

**Pull request merged**